llvm/include/llvm/IR/PatternMatch.h

```diff
@@ -3124,6 +3124,19 @@ inline auto m_c_LogicalOp(const LHS &L, const RHS &R) {
   return m_LogicalOp<LHS, RHS, /*Commutable=*/true>(L, R);
 }
 
+struct GuaranteedNotToBeUndefOrPoison_match {
+  template <typename ITy> bool match(ITy *V) {
+    if (auto *AsValue = dyn_cast<Value>(V))
+      return isGuaranteedNotToBeUndefOrPoison(AsValue);
```

Comment:
I don't like this helper. Some context information (e.g., AC/DT/CxtI) is available in InstCombine.
They are useful to get a more precise analysis result.

Reply:
I'll fix the helper to take caller context.

Reply:
I ended up removing the helper.